ApiReview/API.Polly.Core/NoDocs/Polly.CircuitBreaker/AdvancedCircuitBreakerStrategyOptions.cs

```diff
@@ -0,0 +1,8 @@
+// Assembly 'Polly.Core'
+
+namespace Polly.CircuitBreaker;
+
+public class AdvancedCircuitBreakerStrategyOptions : AdvancedCircuitBreakerStrategyOptions<object>
```

Comment:
Shouldn't we have `AdvancedCircuitBreakerStrategyOptions<TResult> : AdvancedCircuitBreakerStrategyOptions` where the properties stay in the non-generic base class?

Reply:
Polly V8 exposes `Outcome<T>` that encapsulates either result or exception as opposed to V7 that only exposed exceptions for non generic strategies. 

This allowed us to express non-generic strategies simply through options of `object`. Or in other words, the result is either exception or any kind of result. The duplications of options we had in R9 is no longer necessary. 

See this comment for more details about the design:
https://github.com/App-vNext/Polly/issues/1429#issuecomment-1652970413